src/main/java/hudson/remoting/Launcher.java

```diff
@@ -189,6 +189,10 @@ public boolean verify(String s, SSLSession sslSession) {
     @Option(name="-noReconnect",usage="Doesn't try to reconnect when a communication fail, and exit instead")
     public boolean noReconnect = false;
 
+    @Option(name = "-noKeepAlive",
+            usage = "Do not open the socket to the master with SO_KEEPALIVE enabled")
```

Comment:
I would argue that you do not want this unless you know why you do not want it... I could make it a system property rather than a CLI option... should only be required if we have a fundamental misunderstanding of how sockets work...